integration-tests/tests/client/challenges.rs

```diff
@@ -495,7 +495,7 @@ fn test_receive_invalid_chunk_as_chunk_producer() {
         &vec![merkle_paths[0].clone()],
     );
     assert!(env.clients[1]
-        .process_partial_encoded_chunk(MaybeValidated::NotValidated(partial_encoded_chunk))
+        .process_partial_encoded_chunk(MaybeValidated::from(partial_encoded_chunk))
```

Comment:
nit: does it make sense to change this function to be from_non_validated, just to be extra clear?

Reply:
I guess this is a bit philosophical. `from` comes from the `From` trait and gives us `into` as well so using it seems more idiomatic?

Reply:
Yeah, this is bikesheddable. An appropriate std example is that for a long time there weren't `From<T> from Option<T>`, but then it was added at some point. So I'd say it's idiomatic, but not super clear-cut. 

In my personal code style, I tend to prefer inherent factory methods more in general, for a number of weak reasons: 

* uses less of the type-system machinery 
* forces somewhat more readable code. `foo.into()` in particular often is a readability issue, as you need to run the type checker in your head to figure out into *what*. 
* more robust to changes -- in the future, you can add extra parameter to an inherent method

This specific case is clearly borderline though.